### DISCUSSION TITLE
ConfigurationMap update from channel

### DISCUSSION THREAD

**Discussion by bartha-adam** [nextgenhealthcare/connect#5715] [General]
Hi,

Context: we have an app having a UI, communicating with Mirth (over TCP), and we're trying to expose some of the configs from Configuration Map through our custom UI, so the user could be directly editing in the user-facing app.

Envision something like:
1. App sends a custom JSON message to Mirth, including a custom command, like WriteConfig, containing a key-value pair.
    Ex: { "Key"="MyKey", "Value"="MyValue"
2. Mirth Channel receives and validates the request
3. Mirth Channel applies the Key-Value pair to ConfigurationMap
4. Mirth Channels will use the new value from now on (MyValue for MyKey).

It looks like, the Configuration Map cannot be written by JavaScript code from Mirth. Mirth doc also states the same.

And now the question: is there a way to update and apply, the Configuration Map from JavaScript code running inside the Mirth?

All the solutions I think of sound like hacks at this point.

**Comment by bartha-adam** [comment]
Meanwhile, found the use of ConfigurationController getConfigurationProperties/setConfigurationProperties as possible ways to go...

**Comment by pacmano1** [comment]
I'd propose you re-architect a bit. 

Other than perhaps db credentials and one variable that defines "this engine" so to speak in the config map (aka $cfg), store all you configuration information in some db config table that is sourced as needed to the globalMap (aka $g).  Note that $g can hold objects, something $cfg does not do.

IHMO this make most things much easiser and obviously referenceable irrespective to mirth.

**Reply by bartha-adam** [reply]
> I'd propose you re-architect a bit.
> 
> Other than perhaps db credentials and one variable that defines "this engine" so to speak in the config map (aka $cfg), store all you configuration information in some db config table that is sourced as needed to the globalMap (aka $g). Note that $g can hold objects, something $cfg does not do.
> 
> IHMO this make most things much easiser and obviously referenceable irrespective to mirth.

Thanks for the idea, although not sure if it fits our use case.
The idea would be to enable users to configure stuff from our UI but also let users configure directly from Mirth Settings->Configuration Map tab....

**Reply by pacmano1** [reply]
Your call of cousre - but you should wrap change control around all this, no?  Seems easiser to do the entire thing via some UI that updates config tables in DBs rather than have multiple places where something can be updted.
